mule-extensions-api/src/main/java/org/mule/runtime/extension/api/runtime/config/ConfigurationProvider.java

```diff
@@ -34,6 +36,17 @@
    */
   ConfigurationInstance get(Event event);
 
+  /**
+   * Returns a {@link ConfigurationInstance}
+   * <p/>
+   * This method will only return an instance already returned in the past, or {@link Optional#empty()} if an instance was not
+   * generated or has been already removed.
+   *
+   * @param event the event which processing requires the instance
+   * @return a {@link ConfigurationInstance}
+   */
+  Optional<ConfigurationInstance> getIfPresent(Event event);
```

Comment:
not a nice semantic to add, and unnecessary per the comments in the companion PR